src/query/functions/src/scalars/datetime.rs

```diff
@@ -134,6 +137,41 @@ fn int64_domain_to_timestamp_domain<T: AsPrimitive<i64>>(
     })
 }
 
+fn register_convert_timezone(registry: &mut FunctionRegistry) {
+    // 2 arguments function [target_timezone, src_timestamp]
+    registry.register_passthrough_nullable_2_arg::<StringType, TimestampType, TimestampType, _, _>(
+        "convert_timezone",
+        |_, _, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_2_arg::<StringType, TimestampType, TimestampType>(
+            |target_tz, src_timestamp, output, ctx| {
+                // Convert source timestamp from source timezone to target timezone
+                let p_src_timestamp = src_timestamp.to_timestamp(ctx.func_ctx.tz.tz);
+                let src_dst_from_utc = p_src_timestamp.offset().fix().local_minus_utc();
+                let t_tz: Tz = match target_tz.parse() {
+                    Ok(tz) => tz,
+                    Err(e) => {
+                        return ctx.set_error(
+                            output.len(),
+                            format!("cannot parse target `timezone`. {}", e),
+                        );
+                    }
+                };
+
+                let result_timestamp = p_src_timestamp.with_timezone(&t_tz).timestamp_micros();
+                let target_dst_from_utc = p_src_timestamp
+                    .with_timezone(&t_tz)
+                    .offset()
+                    .fix()
+                    .local_minus_utc();
+                let offset_as_micros_sec =
+                    (target_dst_from_utc - src_dst_from_utc) as i64 * MICROS_PER_SEC;
+                let res = result_timestamp.checked_add(offset_as_micros_sec).unwrap();
```

Comment:
is it safe to unwrap here? better to check and return error here.

Reply:
A whole day has 86400000 ms. So I think it's safe unwrap. But it's better add check.

Reply:
Done